services/stats/stats.go

```diff
@@ -202,7 +202,7 @@ func (s *statsdStats) collectPeriodicStats() {
 func (s *statsdStats) Stop() {
 	s.state.clientsLock.RLock()
 	defer s.state.clientsLock.RUnlock()
-	if !s.conf.enabled || !s.state.connEstablished {
+	if !s.conf.enabled.Load() || !s.state.connEstablished {
```

Comment:
Since `enabled` can change to `false` while the stats run, should we always try to close the channels? 

```suggestion
	if !s.state.connEstablished {
```